### ISSUE TITLE
Wrong Object Assignment

### ISSUE THREAD

**Issue by sky9407** [Smorodov/Multitarget-tracker#338]
Hi, I am using your tracker for people counting, and get good results for most of times.
I met this wrong object assignment when people are too close to each other.
Please see this pic below:
![wrong](https://user-images.githubusercontent.com/89692055/131218234-2db80cfa-6b6d-4758-b6d8-39bae03e70c9.jpg)
You can see these two people are very close and they don't move during these three frames.
In frame 1&2, the left person's id is 2, and right person's id is 0; but in frame 3, they exchanged their ids.

Below is my settings:
![config](https://user-images.githubusercontent.com/89692055/131218530-bad3c63b-43eb-4690-a8ea-f696cf28228b.jpg)

We find the m_SPCalculator->Solve function give the wrong assignment.
I am not sure if this is a bug or change track settings can solve this or not? Can you give us some advice? Thanks!

**Comment by Nuzhny007** [comment]
Hi!
Try to change tracking::MatchBipart to tracking::MatchHungarian. I don't tested this algorithm long time and not sure that it works perfectly

**Comment by sky9407** [comment]
Yes, we have changed to MatchHungarian and the test results are right now.
    For MatchBipart, it seems when the currCost is close to m_distThres, and all the weights are 1, the assignments are not correct, so lower the distThres may reduce the frequency of occurrence.
